File: Backend/backend/src/main/java/com/example/backend/entity/SupportTicket.java
package com.example.backend.entity;

import com.example.backend.user.User;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Data
@Entity
@NoArgsConstructor
@Table(name = "support_tickets")
public class SupportTicket {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tenant_id")
    private Tenant tenant;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    private UUID orderId;

    @Column(nullable = false)
    private String subject;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String message;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TicketStatus status = TicketStatus.OPEN;

    @Column(columnDefinition = "TEXT")
    private String adminNotes;

    private Instant createdAt = Instant.now();
    private Instant resolvedAt;

    // Customer escalation to the platform (CraveIt). When a store mishandles or doesn't resolve a complaint,
    // the customer can escalate; escalated tickets become visible to the superadmin for oversight so a store
    // can't quietly mistreat customers (which reflects on CraveIt's reputation).
    @Column(nullable = false)
    private boolean escalated = false;
    private Instant escalatedAt;
    @Column(columnDefinition = "TEXT")
    private String escalationReason;

    public enum TicketStatus {
        OPEN, IN_PROGRESS, RESOLVED, CLOSED
    }
}

File: SuperAdmin/backend/SuperAdmin.API/Models/SupportTicket.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SuperAdmin.API.Models;

// Shared support_tickets table (owned by the Spring app). The SuperAdmin only reads escalated tickets —
// customers escalate to CraveIt when a store mishandles them, giving the platform oversight.
[Table("support_tickets")]
public class SupportTicket
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; }

    [Column("tenant_id")]
    public Guid? TenantId { get; set; }

    [Column("user_id")]
    public Guid? UserId { get; set; }

    [Column("order_id")]
    public Guid? OrderId { get; set; }

    [Column("subject")]
    public string Subject { get; set; } = string.Empty;

    [Column("message")]
    public string Message { get; set; } = string.Empty;

    [Column("status")]
    public string? Status { get; set; }

    [Column("admin_notes")]
    public string? AdminNotes { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [Column("resolved_at")]
    public DateTime? ResolvedAt { get; set; }

    [Column("escalated")]
    public bool Escalated { get; set; }

    [Column("escalated_at")]
    public DateTime? EscalatedAt { get; set; }

    [Column("escalation_reason")]
    public string? EscalationReason { get; set; }

    // Tier-3 + superadmin reply: audience STORE (customer->store) | PLATFORM (store->CraveIt); platformNote
    // is the superadmin's response + when they acted.
    [Column("audience")]
    public string Audience { get; set; } = "STORE";

    [Column("platform_note")]
    public string? PlatformNote { get; set; }

    [Column("platform_reviewed_at")]
    public DateTime? PlatformReviewedAt { get; set; }

    public Tenant? Tenant { get; set; }
    public User? User { get; set; }
}
